### PR TITLE
[1LP][RFR] Version picking unified

### PR DESCRIPTION
**Pull request by quarckster**
Intent
=================

This PR attempts to unify all kinds of version picks in to one `VersionPicker` descriptor.

- `deferred_verpick` renamed to `VersionPicker`
- function `cfme.utils.version.pick` is a method of `VersionPicker`
- `VersionPicker` wraps `widgetastic.utils.VersionPick`
- removed unneeded version picks
- removed `cfme/automate/dialog_box.py`, `cfme/automate/dialog_element.py`, `cfme/automate/dialog_tab.py` and `cfme/automate/service_dialog.py`. They are not needed anymore. Versions of these files for CFME5.9 and newer are in `cfme/automate/dialogs`

Replacing `widgetastic.utils.VersionPick` by `cfme.utils.version.VersionPicker` follows up in the next PR.

{{pytest: -v -k "test_infra_provider_crud or test_snapshot_crud" --long-running}}

### CODE REVIEW

File: cfme/common/provider.py [hunk 1]
```diff
@@ -532,8 +536,10 @@ def delete(self, cancel=True):
             cancel: Whether to cancel the deletion, defaults to True
         """
         view = navigate_to(self, 'Details')
-        item_title = version.pick({'5.9': 'Remove this {} Provider from Inventory',
-                                   version.LOWEST: 'Remove this {} Provider'})
+        item_title = VersionPicker({
```
Comment (RonnyPfannschmidt) [inline cfme/common/provider.py]: **required** there seem to be inconsistencies in picking 5.9 vs lowest, some being removed, some being added - we should have this uniformly

**Comment by dajoRH** [comment]
Lint report for commit 09e02090f213c67e2c6da8c62dadf8915e642fd8:

`cfme/tests/containers/test_properties.py`:
- :red_circle: Line 20:1: **F401** *'cfme.utils.version.VersionPicker' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 36098bb1789d12b02b336ee69888de6e12216c0d:

`cfme/control/explorer/conditions.py`:
- :heavy_exclamation_mark: Line 336:50: **W291** *trailing whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/automate/test_service_dialog.py [hunk 2]
```diff
@@ -44,9 +43,7 @@ def test_crud_service_dialog(appliance):
     dialog, element = create_dialog(appliance, element_data)
     view_cls = navigator.get_class(element.parent, 'Add').VIEW
     view = element.appliance.browser.create_view(view_cls)
-    flash_message = version.pick({
-        '5.8': 'Dialog "{}" was added'.format(dialog.label),
-        '5.9': '{} was saved'.format(dialog.label)})
```
Comment (sshveta) [inline cfme/tests/automate/test_service_dialog.py]: Question : Are we removing support for 5.8 ?

File: cfme/utils/appliance/__init__.py [hunk 3]
```diff
@@ -1035,12 +1035,11 @@ def miqqe_patch_applied(self):
 
     @logger_wrap("Patch appliance with MiqQE js: {}")
     def patch_with_miqqe(self, log_callback=None):
-
         # (local_path, remote_path, md5/None) trio
-        autofocus_patch = pick({
+        autofocus_patch = VersionPicker({
             '5.5': 'autofocus.js.diff',
             '5.7': 'autofocus_57.js.diff'
-        })
```
Comment (sshveta) [inline cfme/utils/appliance/__init__.py]: we can remove 5.5 and 5.7 version picks now

File: cfme/automate/dialog_collection_pick.py [hunk 4]
```diff
@@ -1,9 +0,0 @@
-from widgetastic.utils import VersionPick
-
-from cfme.automate.service_dialogs import DialogCollection as dc58  # noqa
-from cfme.automate.dialogs.service_dialogs import DialogCollection as dc59  # noqa
-from cfme.utils import version
-
-collection_pick = VersionPick({
-    version.LOWEST: dc58,
-    '5.9': dc59})
```
Comment (sshveta) [inline cfme/automate/dialog_collection_pick.py]: There are dialog code files for both 5.8 and 5.9 that gets picked by this code . If we are dropping automation for 5.8 then we will have to remove those files too .

File: cfme/utils/appliance/__init__.py (hunk 3 above)
Comment (quarckster) [reply cfme/utils/appliance/__init__.py]: Actually, this method is not needed anymore, but it should be removed in a next PR.

File: cfme/automate/dialog_collection_pick.py (hunk 4 above)
Comment (quarckster) [reply cfme/automate/dialog_collection_pick.py]: Done

Comment (sshveta) [reply cfme/automate/dialog_collection_pick.py]: @quarckster : Also please remove automate/dialog_tab.py , dialog_element.py and dialog_box.py

File: cfme/utils/appliance/__init__.py (hunk 3 above)
Comment (sshveta) [reply cfme/utils/appliance/__init__.py]: :+1:

File: cfme/tests/automate/test_service_dialog.py (hunk 2 above)
Comment (sshveta) [reply cfme/tests/automate/test_service_dialog.py]: :+1:

File: cfme/automate/dialog_collection_pick.py (hunk 4 above)
Comment (quarckster) [reply cfme/automate/dialog_collection_pick.py]: Done

**Comment by psav** [comment]
There seem to be a whole lot of files removed without explanation - please correct me if I'm wrong

**Comment by quarckster** [comment]
@psav deleted files are used only in 5.8, cfme/automate/dialogs/dialog_* are for 5.9 and newer and  cfme/automate/dialog_* are for 5.8.

**Pull request merged**